Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/J2ObjCParser.kt

```diff
@@ -0,0 +1,218 @@
+        encoding = "[]", //TODO: Implement encoding properly
+        parameters = methodParameters,
+        returnType = methodReturnType,
+        isVariadic = false,
+        isClass = access == Opcodes.ACC_PUBLIC + Opcodes.ACC_STATIC, // TODO: Currently only handles Public instance and Public static methods, true when static, false when instance
+        nsConsumesSelf = false,
+        nsReturnsRetained = false,
+        isOptional = false,
+        isInit = false,
+        isExplicitlyDesignatedInitializer = false)
+    }
+  }
+
+  /**
+   * Parses an ASM method's parameters and returns a list of Kotlin parameters
+   *
+   * @param method ASM method descriptor
+   * @return A list of Kotlin parameters with associated types/names
+   */
+
+  private fun parseMethodParameters(methodDesc: String, paramNames: List<String>): List<Parameter> {
+    val methodParameters = mutableListOf<Parameter>()
+    val parameterTypes = getArgumentTypes(methodDesc)
+
+    for (i in 0 until parameterTypes.size) {
+      val paramName = paramNames.get(i)
+      when (parameterTypes.get(i).className) {
+        "boolean" -> methodParameters.add(Parameter(name = paramName, type = parseType(parameterTypes.get(i)), nsConsumed = false))
+        "byte" -> methodParameters.add(Parameter(name = paramName, type = parseType(parameterTypes.get(i)), nsConsumed = false))
+        "char" ->  methodParameters.add(Parameter(name = paramName, type = parseType(parameterTypes.get(i)), nsConsumed = false))
+        "double" ->  methodParameters.add(Parameter(name = paramName, type = parseType(parameterTypes.get(i)), nsConsumed = false))
+        "float" ->  methodParameters.add(Parameter(name = paramName, type = parseType(parameterTypes.get(i)), nsConsumed = false))
+        "int" -> methodParameters.add(Parameter(name = paramName, type = parseType(parameterTypes.get(i)), nsConsumed = false))
+        "long" ->  methodParameters.add(Parameter(name = paramName, type = parseType(parameterTypes.get(i)), nsConsumed = false))
+        "short" ->  methodParameters.add(Parameter(name = paramName, type = parseType(parameterTypes.get(i)), nsConsumed = false))
+        else -> {
+          throw NotImplementedError("Have not implemented this type yet: " + parameterTypes.get(i).className)
+        }
+      }
+    }
+    return methodParameters
+  }
+
+  /**
+   * Parses an ASM method's return type and returns a Kotlin type
+   *
+   * @param method ASM method descriptor
+   * @return Type corresponding to method's return type
+   */
+  private fun parseMethodReturnType(methodDesc: String): Type {
+    return parseType(getReturnType(methodDesc))
+  }
+
+  /**
+   * Helper function to parse ASM types and return a Kotlin type
+   *
+   * @param type ASM type
+   * @return Kotlin type
+   */
+  private fun parseType(type: org.jetbrains.org.objectweb.asm.Type): Type {
+    when (type.className) {
```

Comment:
more idiomatic syntax for this is to use when as an expression:

return when (type.className) {
  "boolean" -> ObjCBoolType
  "byte" -> ...
}

Reply:
Done.